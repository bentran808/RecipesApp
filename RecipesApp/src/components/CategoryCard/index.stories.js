import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import { category } from 'mocks';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import CategoryCard from '.';

storiesOf('Category Card', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => <CategoryCard item={category} onPressCategory={action('clicked-card')} />);
