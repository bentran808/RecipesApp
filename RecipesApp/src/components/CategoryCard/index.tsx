import React, { useCallback } from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

type Props = {
  item: Category;
  onPressCategory: (item: Category) => void;
};

const CategoryCard = ({ item, onPressCategory }: Props) => {
  const handlePress = useCallback(() => {
    onPressCategory(item);
  }, []);

  return (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.categoriesItemContainer}>
        <Image style={styles.categoriesPhoto} source={{ uri: item.photo_url }} />
        <Text style={styles.categoriesName}>{item.name}</Text>
        <Text style={styles.categoriesInfo}>{item.recipes.length} recipes</Text>
      </View>
    </TouchableOpacity>
  );
};

export default CategoryCard;
