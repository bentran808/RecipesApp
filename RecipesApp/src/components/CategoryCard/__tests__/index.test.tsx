import CategoryCard from 'components/CategoryCard';
import { category } from 'mocks';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

describe('Category Card Component', () => {
  const onPressCategory = jest.fn();
  test('should render correctly', () => {
    const tree = renderer
      .create(<CategoryCard item={category} onPressCategory={onPressCategory} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePress', () => {
    const component = renderer.create(
      <CategoryCard item={category} onPressCategory={onPressCategory} />
    );
    const button = component.root.findAllByType(TouchableOpacity)[0];

    button.props.onPress();

    expect(onPressCategory).toHaveBeenCalled();
  });
});
