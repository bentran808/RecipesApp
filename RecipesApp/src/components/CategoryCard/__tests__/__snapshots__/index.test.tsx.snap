// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Card Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 20,
        "borderWidth": 0.5,
        "flex": 1,
        "height": 215,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.telegraph.co.uk/content/dam/Travel/2019/January/france-food.jpg?imw'id'th=1400",
        }
      }
      style={
        Object {
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderRadius": 20,
          "height": 155,
          "shadowColor": "blue",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 5,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#333",
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 8,
          "textAlign": "center",
        }
      }
    >
      Cookies
    </Text>
    <Text
      style={
        Object {
          "marginBottom": 5,
          "marginTop": 3,
        }
      }
    >
      0
       recipes
    </Text>
  </View>
</View>
`;
